filesink/store.go

```diff
@@ -53,6 +104,19 @@ func (c S3StoreConfig) Validate() error {
 		}
 	}
 
+	if c.Credentials == nil {
+		if c.AWSAccessKeyID == "" {
+			return errors.New("missing 'awsAccessKeyId'")
+		}
+		if c.AWSSecretAccessKey == "" {
+			return errors.New("missing 'awsSecretAccessKey'")
+		}
+	} else {
```

Comment:
nit: Could simplify this a tiny bit, using `else if err := c.Credentials.Validate(); err != nil { ...}`